org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/customservice/synapse/dynamic/db/DatabaseService.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.eclipse.lemminx.customservice.synapse.dynamic.db;
+
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class DatabaseService {
+    private static final Logger LOGGER = Logger.getLogger(DatabaseService.class.getName());
+
+    private static final String FIELD_TYPE_ATTRIBUTE = "attribute";
+    private static final String DYN_PARAM_PREFIX = "dyn_param_";
+    private static final String IS_NULLABLE_COLUMN = "IS_NULLABLE";
+    private static final String NULLABLE_NO = "NO";
+    private static final String BOOLEAN_TRUE = "true";
+    private static final String BOOLEAN_FALSE = "false";
+    private static final String COLUMN_NAME_COLUMN = "COLUMN_NAME";
+    private static final String TYPE_NAME_COLUMN = "TYPE_NAME";
+
+    /**
+     * @param connectionUrl
```

Comment:
Update the description with the method usecase and the describe the method parameters